src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let stateForResult = { ...state };
+
+  for (const action in actions) {
+    if (actions[action].type === 'addProperties') {
+      const addValues = actions[action].extraData;
+
+      for (const addValue in addValues) {
+        stateForResult[addValue] = addValues[addValue];
+      }
+      result.push(stateForResult);
```

Comment:
you can push only one time after all if's

Reply:
@etojeDenys 
![image](https://github.com/mate-academy/js_stateful-clones/assets/107351339/6bd8fb80-6c9e-4bd9-8d5b-3756b0fce278)